### PR TITLE
Special radio channels now use bitflags instead of individual vars

### PR DESCRIPTION
**Pull request by Rhials**
## About The Pull Request

Radios/encryption keys now use a `RADIO_SPECIAL_XXXXX` bitflag for behaviors related to "special" radio channels (Binary, Centcom, Syndie).

I decided to do this because I wanted to add a radio channel for pirates and hunters (hence the branch name), but it felt weird adding two more variables. The more I look at the changes I've made here the more I realize that the effort was probably not worth the utility but whatever.

This also subtypes some varedited intercoms and makes them their own objects.
## Why It's Good For The Game

Compresses a whopping three (3!) variables into a single one. Easier to scale (I guess?). I felt like adding a fourth/fifth variable and just moving on with the original project, but decided "lets do this the unnecessarily hard way instead".
## Changelog
:cl: Rhials
code: Radios/encryption keys now use a single variable for "special" frequencies. Please report if you experience any strangeness with accessing/being unable to access the Centcom, Syndicate, or Cyborg radio.
/:cl:

### CODE REVIEW

**Comment by Rhials** [comment]
Well, it looks like it's passing the unit tests. I'd like an extra careful review of my last commit because I feel like I've messed up my understanding of bitwise operators somewhere in here.

File: code/game/objects/items/devices/radio/radio.dm [hunk 1]
```diff
@@ -178,7 +167,7 @@
 /obj/item/radio/proc/make_syndie() // Turns normal radios into Syndicate radios!
 	qdel(keyslot)
 	keyslot = new /obj/item/encryptionkey/syndicate()
-	syndie = TRUE
+	special_channels &= RADIO_SPECIAL_SYNDIE
```
Comment (AnturK) [inline code/game/objects/items/devices/radio/radio.dm:170]: |=

File: code/game/objects/items/devices/radio/radio.dm [hunk 2]
```diff
@@ -335,7 +324,7 @@
 		channel = null
 
 	// Nearby active jammers prevent the message from transmitting
-	if(is_within_radio_jammer_range(src) && !syndie)
+	if(is_within_radio_jammer_range(src) && special_channels & RADIO_SPECIAL_SYNDIE)
```
Comment (AnturK) [inline code/game/objects/items/devices/radio/radio.dm:327]: Missing !

File: code/modules/admin/verbs/borgpanel.dm [hunk 3]
```diff
@@ -153,25 +153,25 @@ ADMIN_VERB(borg_panel, R_ADMIN, "Show Borg Panel", ADMIN_VERB_NO_DESCRIPTION, AD
...
+						borg.radio.keyslot.special_channels &= ~RADIO_SPECIAL_CENTCOM
 				message_admins("[key_name_admin(user)] removed the [channel] radio channel from [ADMIN_LOOKUPFLW(borg)].")
 				log_silicon("[key_name(user)] removed the [channel] radio channel from [key_name(borg)].")
 			else // We're adding a channel
 				if (!borg.radio.keyslot) // Assert that an encryption key exists
 					borg.radio.keyslot = new()
 				borg.radio.keyslot.channels[channel] = 1
 				if (channel == RADIO_CHANNEL_SYNDICATE)
-					borg.radio.keyslot.syndie = TRUE
+					borg.radio.keyslot.special_channels &= RADIO_SPECIAL_SYNDIE
```
Comment (AnturK) [inline code/modules/admin/verbs/borgpanel.dm:172]: |=

File: code/modules/admin/verbs/borgpanel.dm [hunk 4]
```diff
@@ -153,25 +153,25 @@ ADMIN_VERB(borg_panel, R_ADMIN, "Show Borg Panel", ADMIN_VERB_NO_DESCRIPTION, AD
...
 				else if (channel == "CentCom")
-					borg.radio.keyslot.independent = TRUE
+					borg.radio.keyslot.special_channels &= RADIO_SPECIAL_CENTCOM
```
Comment (AnturK) [inline code/modules/admin/verbs/borgpanel.dm:174]: |=

File: code/modules/capture_the_flag/ctf_classes.dm [hunk 5]
```diff
@@ -55,7 +55,7 @@
 		var/obj/item/radio/headset = human_to_equip.ears
 		headset.set_frequency(team_radio_freq)
 		headset.freqlock = RADIO_FREQENCY_LOCKED
-		headset.independent = TRUE
+		headset.special_channels &= RADIO_SPECIAL_CENTCOM
```
Comment (AnturK) [inline code/modules/capture_the_flag/ctf_classes.dm:58]: |=

**Review by AnturK — CHANGES_REQUESTED** [review]
A few operator errors

File: code/modules/pai/say.dm [hunk 6]
```diff
@@ -1,2 +1,4 @@
 /mob/living/silicon/pai/binarycheck()
-	return radio?.translate_binary
+	if(radio)
+		return (radio.special_channels & RADIO_SPECIAL_BINARY)
+	return FALSE
```
Comment (Ghommie) [inline code/modules/pai/say.dm:4]:
```suggestion
	return (radio?.special_channels & RADIO_SPECIAL_BINARY)
```

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Pull request merged**